mantis-server/mantis-server-worker/src/main/java/io/mantisrx/server/worker/MantisWorker.java

```diff
@@ -121,27 +125,33 @@ public void run() {
             @Override
             public void start() {
                 executeStageSubject
-                    .asObservable()
-                    .first()
-                    .subscribe(wrappedRequest -> {
-                        task = new Task(
-                            wrappedRequest,
-                            config,
-                            gateway,
-                            ClassLoaderHandle.fixed(getClass().getClassLoader()),
-                            SinkSubscriptionStateHandler
-                                .Factory
-                                .forEphemeralJobsThatNeedToBeKilledInAbsenceOfSubscriber(
-                                    gateway,
-                                    Clock.systemDefaultZone()),
-                            Optional.empty(),
-                            jobToRun);
-                        taskStatusUpdateSubscription =
-                            task
-                                .getStatus()
-                                .subscribe(statusUpdateHandler::onStatusUpdate);
-                        task.startAsync();
-                    });
+                        .asObservable()
+                        .first()
+                        .subscribe(wrappedRequest -> {
+                            try {
+                                task = new Task();
+                                task.setWrappedExecuteStageRequest(wrappedRequest);
+                                task.setWorkerConfiguration(config);
+                                task.setMantisMasterGateway(gateway);
+                                task.setUserCodeClassLoader(ClassLoaderHandle.createUserCodeClassloader(
+                                    wrappedRequest.getRequest(),
+                                    ClassLoaderHandle.fixed(getClass().getClassLoader())));
+                                task.setSinkSubscriptionStateHandlerFactory(SinkSubscriptionStateHandler
+                                    .Factory
+                                    .forEphemeralJobsThatNeedToBeKilledInAbsenceOfSubscriber(
+                                        gateway,
+                                        Clock.systemDefaultZone()));
+                                task.setJob(jobToRun);
+
+                                taskStatusUpdateSubscription =
+                                    task
+                                        .getStatus()
+                                        .subscribe(statusUpdateHandler::onStatusUpdate);
```

Comment:
The task also needs to expose `vmTaskStatusSubject` which will have to be fed into the mesos executor callback handler. https://github.com/Netflix/mantis/blob/master/mantis-server/mantis-server-worker/src/main/java/io/mantisrx/server/worker/MantisWorker.java#L155-L157